packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -178,6 +215,28 @@ const processOnEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
         }
       }
 
+      // Determine if table needs tags update.
+      // For Gen2 deployments, the tags do not change between deployments. This check is to ensure that
+      // the tags are applied to tables created earlier (before adding tagging support).
+      // TODO: Handle tags update for CDK construct deployments
```

Comment:
Can't comment directly on the relevant code above, but we should update our GSI tests to ensure that tags propagate correctly during GSI updates that entirely replace the table.